include/pmtv/type_helpers.hpp

```diff
@@ -36,11 +36,17 @@ struct as_pmt<TemplateType, std::tuple<T...>> {
 template<template<typename... > class VariantType, class... Args>
 using as_pmt_t = typename detail::as_pmt<VariantType, Args...>::type;
 
+static constexpr bool support_size_t = !std::is_same_v<std::size_t, uint16_t> && !std::is_same_v<std::size_t, uint32_t> && !std::is_same_v<std::size_t, uint64_t>;
```

Comment:
So I think that the idea here is as follows:
- On Windows, `size_t` is the same type as `uint64_t`.  So that will be treated the same.
- On linux, `size_t` is a different type that will be handled by adding it to the pmt definition.

In both cases we treat `size_t` like a `uint64_t`.

Can you add some comments to explain what is going on?  It took me a few minutes to figure it out.

Reply:
Sure, I will add comments.